BCF-3015 Add Configuration for Encoding

EasterTheBunny opened the pull request:
Solana allows multiple encoding types for account data. This commit adds support for
two encoding types `borsh` and `bincode`.

reductionista commented on pkg/solana/config/chain_reader_test.go:
```diff
@@ -0,0 +1,117 @@
+		t.Parallel()
+
+		var result config.ChainReader
+		require.NotNil(t, json.Unmarshal([]byte(invalidJSON), &result))
```
I was staring at this thinking "how can this be working with `require.NotNil`?" then realized this is the line the CI failure happened on!

```suggestion
		_, err = json.Unmarshal([]byte(invalidJSON), &result))
		assert.Error(t, err)
```

reductionista replied:
(or `assert.ErrorIs` with whatever error type it's supposed to return?)

reductionista commented on pkg/solana/config/chain_reader_test.go:
```diff
@@ -0,0 +1,117 @@
+
+	_, err := json.Marshal(config.EncodingType(100))
+
+	require.NotNil(t, err)
```
```suggestion
	require.ErrorIs(t, err, types.ErrInvalidConfig)
```

reductionista commented on pkg/solana/config/chain_reader_test.go:
```diff
@@ -0,0 +1,117 @@
+	var tp config.EncodingType
+
+	require.NotNil(t, json.Unmarshal([]byte(`42`), &tp))
+	require.NotNil(t, json.Unmarshal([]byte(`"invalid"`), &tp))
```
These must be failing also? See above

EasterTheBunny replied:
I botched the rebase. :(

reductionista commented:
Looks good to me, aside from the `NotNil` errors I commented on, and the undefined `namespaceBindings` issue

EasterTheBunny committed:
Add Configuration for Encoding

Solana allows multiple encoding types for account data. This commit adds support for
two encoding types `borsh` and `bincode`.
